sdk/feature-management-applicationinsights-browser/src/telemetry.ts

```diff
@@ -12,16 +12,48 @@ import { IEventTelemetry } from "@microsoft/applicationinsights-web";
  */
 export function createTelemetryPublisher(client: ApplicationInsights): (event: EvaluationResult) => void {
     return (event: EvaluationResult) => {
+        if (event.feature === undefined) {
+            return;
+        }
+
         const eventProperties = {
-            "FeatureName": event.feature?.id,
+            "Version": EVALUATION_EVENT_VERSION,
+            "FeatureName": event.feature.id,
             "Enabled": event.enabled.toString(),
             // Ensure targetingId is string so that it will be placed in customDimensions
             "TargetingId": event.targetingId?.toString(),
             "Variant": event.variant?.name,
             "VariantAssignmentReason": event.variantAssignmentReason,
         };
 
-        const metadata = event.feature?.telemetry?.metadata;
+        if (event.feature.allocation?.default_when_enabled) {
+            eventProperties["DefaultWhenEnabled"] = event.feature.allocation.default_when_enabled;
+        }
+
+        if (event.variantAssignmentReason === VariantAssignmentReason.DefaultWhenEnabled) {
+            let percentileAllocationPercentage = 0;
+            if (event.variant !== undefined && event.feature.allocation !== undefined && event.feature.allocation.percentile !== undefined) {
+                for (const percentile of event.feature.allocation.percentile) {
+                    if (percentile.variant === event.variant.name) {
+                        percentileAllocationPercentage += percentile.to - percentile.from;
+                    }
+                }
+            }
+            eventProperties["PercentileAllocationPercentage"] = (100 - percentileAllocationPercentage).toString();
```

Comment:
the percentage seems to include percentrage of all other unassigned variants, please confirm the design

Reply:
I referenced the old pr. There was a bug. It is fixed in the latest PR: https://github.com/microsoft/FeatureManagement-Python/pull/45